### PR TITLE
Profile population

### PR DESCRIPTION
**Pull request by owensimpson**
Resolves [Populate My Profile](https://trello.com/c/baEyqpaO)

### Changes:
- Users profile pages will now populate when visiting after saving information previously. This is done by adding a trpc query to the profile page and changing the state of the components
- Added a two new types for type safety

### Contributors
Worked on by Luke and Owen with help from David

### CODE REVIEW

**Comment by owensimpson** [comment]
Note: the failing tests are due to timezones, which are fixed in the other branch

File: src/utils/types.ts [hunk 1]
```diff
@@ -44,3 +45,9 @@ export type PublicUser = {
 
 export type User = inferQueryOutput<"user.me">;
 export type GeoJsonUsers = inferQueryOutput<"mapbox.geoJsonUsersList">;
+
+export type CarpoolAddress = {
+  place_name: string;
+  center: [longitude: number, latitude: number];
+};
+export type CarpoolFeature = Feature & CarpoolAddress;
```
Comment (devashish9) [inline src/utils/types.ts:53]: What does "&" do here exactly?

Comment (owensimpson) [reply src/utils/types.ts:53]: CarpoolFeature conforms to both CarpoolAddress and the external Feature type. Essentially, it has the properties of both types.

File: src/pages/profile.tsx [hunk 2]
```diff
@@ -129,30 +115,72 @@ const Profile: NextPage = () => {
     },
     resolver: zodResolver(onboardSchema),
   });
+  const { data: user } = trpc.useQuery(["user.me"], { refetchOnMount: true });
```
Comment (devashish9) [inline src/pages/profile.tsx:118]: what does refetchOnMount do?

File: src/pages/profile.tsx [hunk 3]
```diff
@@ -129,30 +115,72 @@ const Profile: NextPage = () => {
...
+    setStartAddressSelected({
+      place_name: user.startAddress,
+      center: [user.startCoordLng, user.startCoordLat],
+    });
+    setCompanyAddressSelected({
+      place_name: user.companyAddress,
+      center: [user.companyCoordLng, user.companyCoordLat],
+    });
+
+    reset({
```
Comment (devashish9) [inline src/pages/profile.tsx:164]: what does reset do?

File: src/pages/profile.tsx [hunk 4]
```diff
@@ -437,16 +471,29 @@ const Profile: NextPage = () => {
...
+                        }) => (
+                          <Checkbox
+                            key={day + index.toString()}
+                            sx={{
+                              input: { width: 1, height: 1 },
+                              padding: 0,
+                            }}
+                            checked={value}
+                            onChange={onChange}
+                            checkedIcon={<DayBox day={day} isSelected={true} />}
```
Comment (devashish9) [inline src/pages/profile.tsx:490]: what's the difference between the checked field and checkedIcon?

File: src/pages/profile.tsx [hunk 5]
```diff
@@ -437,16 +471,29 @@ const Profile: NextPage = () => {
...
+                            icon={<DayBox day={day} isSelected={false} />}
```
Comment (devashish9) [inline src/pages/profile.tsx:491]: and then ig what is this icon field?

**Review by devashish9 — APPROVED** [review]
LGTM. Just please answer my questions before you merge it in

File: src/pages/profile.tsx (hunk 2 above)
Comment (owensimpson) [reply src/pages/profile.tsx:118]: This was Luke's solution to the page not re-rendering when a user navigated to it using the back arrows after saving information iirc

File: src/pages/profile.tsx (hunk 3 above)
Comment (owensimpson) [reply src/pages/profile.tsx:164]: Reset is pulled from our form inputs, and resets the default state of the components to include the user.me query

File: src/pages/profile.tsx (hunk 4 above)
Comment (owensimpson) [reply src/pages/profile.tsx:490]: @lukejianu

File: src/pages/profile.tsx (hunk 5 above)
Comment (owensimpson) [reply src/pages/profile.tsx:491]: I believe checkedIcon vs icon represents the two possible states of the icon. This was already here if I remember correctly.

**Pull request merged**